x-pack/plugin/esql/qa/testFixtures/src/main/resources/math.csv-spec

```diff
@@ -988,3 +988,56 @@ row d = 1/0.0 | eval s = is_infinite(sqrt(d));
 d:double | s:boolean
 Infinity | true
 ;
+
+least
+// tag::least[]
+ROW a = 10, b=20
+| EVAL l = LEAST(a, b);
+// end::least[]
+
+// tag::least-result[]
+a:integer | b:integer | l:integer
+       10 |        20 | 10
+// end::least-result[]
+;
+
+leastNull
+ROW l=LEAST(10, 5, null);
+
+l:integer
+null
+;
+
+leastMany
+ROW l=LEAST(10, 5, 1, -100, 0, 1234, -10000);
+
+l:integer
+-10000
+;
+
+
+greatest
+// tag::greatest[]
+ROW a = 10, b=20
+| EVAL g = GREATEST(a, b);
+// end::greatest[]
+
+// tag::greatest-result[]
+a:integer | b:integer | g:integer
+       10 |        20 | 20
+// end::greatest-result[]
+;
+
+greatestNull
+ROW g=GREATEST(10, 5, null);
+
+g:integer
+null
+;
+
+greatestMany
+ROW g=GREATEST(10, 5, 1, -100, 0, 1234, -10000);
+
+g:integer
+1234
+;
```

Comment:
question/nit: do we allow varargs functions to take 0 arguments? IMHO, a test case for this would be good, either as int test (if it's allowed) or as unit test (if it's supposed to error out).

Reply:
I'm adding some more tests asserting that we don't support 0 arguments. Actually, none of our varargs functions should allow 0 arguments. That requires a bit of fiddling on the constructors which I'll do.